torch/_dynamo/symbolic_convert.py

```diff
@@ -234,16 +244,27 @@ def inner(self: "InstructionTranslatorBase", inst: Instruction):
             if value.is_python_constant() and bool(value.as_python_constant()):
                 self.jump(inst)
                 return
-
-            # Manually insert torch._assert instead of python assert and jump over
-            # assert related instructions as we don't need them anymore.
-            self.output.create_proxy(
-                "call_function",
-                torch._assert,
-                *proxy_args_kwargs((value, error_msg), {}),
-            )
-            self.jump(inst)
-            return
+            # To avoid DataDependentOutputException when running node with fake tensor,
+            # need to ensure no data-dependent before rewriting assert with torch._assert
+            if (
+                isinstance(value, TensorVariable)
+                and hasattr(self.output.compiler_fn, "__name__")
+                and self.output.compiler_fn.__name__ == "inductor"
```

Comment:
Why is it specific to "inductor" backend?

Reply:
It should not be specific to "inductor" backend. I updated code in and reopened a new PR https://github.com/pytorch/pytorch/pull/94895 because my local ghstack does not work now.